src/Pact/Types/Term.hs

```diff
@@ -1214,11 +1207,31 @@ pattern TLitInteger i <- TLiteral (LInteger i) _
 pattern TLitBool :: Bool -> Term t
 pattern TLitBool b <- TLiteral (LBool b) _
 
-
 tLit :: Literal -> Term n
 tLit = (`TLiteral` def)
 {-# INLINE tLit #-}
 
+$( return [] )
+instance Eq n => Ord (Def n) where
+  a `compare` b = nm a `compare` nm b
+    where nm d = (_dModule d, _dDefName d)
+
+deriving instance Show n => Show (Def n)
+deriving instance Eq n => Eq (Def n)
+
+deriving instance Show n => Show (Object n)
+deriving instance Eq n => Eq (Object n)
+
+deriving instance Show n => Show (Term n)
+deriving instance Eq n => Eq (Term n)
+
+deriving instance Eq d => Eq (Ref' d)
+deriving instance Show d => Show (Ref' d)
+
+deriving instance Eq FunApp
+deriving instance Show FunApp
```

Comment:
You'll have to explain why there were moved down here and what that TH is doing

Reply:
There is a link in the commit message: 
https://downloads.haskell.org/~ghc/8.10.1/docs/html/users_guide/8.10.1-notes.html#template-haskell

Reply:
It's ugly, but, well, in the end TH is nothing more than glorified CPP. It's just code generation and the semantics are mostly determined by the internal operations of the compiler. Unfortunately, any nontrivial use tents to cause trouble.

Reply:
Okay, but that still doesn't tell me why these were moved from working deriving clauses to standalone deriving? The `Eq` and `Show` instances shouldn't have been affected by this.

Reply:
If I remember correctly what the problem was. I think, there is some recursion going on between the `Show` (`Eq`) and `Show1` (`Eq1`) instance. But I have to admit that I don't remember. I'll I remember is that this works and the the `$(return [])` statement is crucial for it :-(

Just try it out yourself, if you like puzzles :-)